vumi/transports/smpp/protocol.py

```diff
@@ -598,6 +598,12 @@ def submit_csm_udh(self, vumi_message_id, destination_addr, **pdu_params):
 
         pdu_params = pdu_params.copy()
         split_msg = self.csm_split_message(pdu_params.pop('short_message'))
+        if len(split_msg) == 1:
+            # only 1 part, so just send the message without the UDH
+            sequence_number = yield self.submit_sm(
+                vumi_message_id, destination_addr, short_message=split_msg[0],
+                **pdu_params)
+            returnValue([sequence_number])
```

Comment:
I think `sequence_number` might already be a list; `extend()` is used on the result of this call below, and the current vumi code treats it as a least here as well:

https://github.com/praekeltfoundation/vumi/blob/02518583774bcb4db5472aead02df617e1725997/vumi/transports/smpp/smpp_service.py#L376-L381

Reply:
Excellent catch! Yes. And my tests missed my error because of inappropriate copy-paste :/